dotnet/src/Agents/Framework/AgentException.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System;
+
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// Agent specific <see cref="KernelException"/>.
+/// </summary>
+public class AgentException : KernelException
```

Comment:
We moved away from having an exception for each SK component and decided to go with a minimal number of custom exceptions, following the principle of "less is more." and add them later if they are really necessary. I wonder in which scenarios the agent exception can be useful to agent callers, considering that it does not add anything on top of the inherited kernel exception.

Reply:
We already have `AgentException` in the experimental framework.  imho - Only throwing a single leaf exception type is a .Net & Java anti-pattern.  Since it inherits from KernelException, that's good...but it also allows for finer discrimintation in mixed app code.

https://github.com/microsoft/semantic-kernel/issues/5756

Reply:
I don't love that we only have a single exception in SK, I find that it makes programmatically handling details of the issue challenging.  However, I also think it's weird to have one place that breaks the pattern and has its own exception, especially since it's not adding functionality to KernelException. I recommend that we stick with the existing pattern unless we have a really compelling reason to break it.

As a separate issue and PR, I would love if we could holistically improve the granularity of our exceptions in a non-breaking way.

Reply:
Totally agree.  They can still all be `KernelException`...I actually reached out to Sergey last year when I saw this happening.  Text parsing exception messages just doesn't scale.

Reply:
> I don't love that we only have a single exception in SK, I find that it makes programmatically handling details of the issue challenging. 

Nothing prevents us from adding new exceptions to SK. Having only one exception was never the goal. The objective was to have one exception for the majority of cases and add more exceptions if additional and actionable details are truly necessary. So, if we find that SKException is missing some crucial information that needs to be expressed through exception properties, let's consider introducing another exception that would include those. As always, we should follow the "less is more" principle, because it's easy to add exceptions but difficult to modify or remove them without causing breaking changes in the open-source world.

> Totally agree. They can still all be KernelException...I actually reached out to Sergey last year when I saw this happening. Text parsing exception messages just doesn't scale.

The parsing of messages was never a goal. If we find ourselves doing it, we should stop. I thought I conveyed this during our conversation at that time. - _"The idea is simple - minimize number of custom SK exceptions by replacing them with SKException, use standard .NET exceptions where possible, and add derivatives of SKException with typed properties having additional details(**no string parsing**) only and only in case SK or SK consumer code needs to make a decision(resiliency, fallback, etc) based on exception details."_